engine/src/tangl/core38/runtime_op.py

```
@@ -0,0 +1,167 @@
+        >>> RuntimeOp.apply_all("print(f'before: {abc}')", "abc = 456",
+        ...                     "print(f'after: {abc}')", ns={'abc': 123})
+        before: 123
+        after: 456
+        {'abc': 456}
+    """
+    expr: str
+
+    @classmethod
+    def _eval_expr(cls, s: str, ns: StringMap = None) -> Any:
+        ns = ns or {}
+        return eval(s, safe_builtins, ns)
```

Ruff (0.15.0)

[error] Use of possibly insecure function; consider using `ast.literal_eval`

(S307)